### PR TITLE
Sample implementation of Productivity Engineer take-home programming task - 202106 - Oncall Calendar Creation

### PR DESCRIPTION
**Pull request by BigLep**
This is an exmaple of biglep@ doing [Productivity Engineer take-home programming task - 202106 - Oncall Calendar Creation](https://docs.google.com/document/d/1bA9YJV5b71_d2Pga-vZwo79nE26arFuMxVt0LHSm044/edit)

A rendered view of the notebook can be found [here](https://github.com/protocol/web3-dev-team/blob/Oncall-Calendar-Creation-takehome-programming-assignment/interviewing/take-home-assignments/oncall-calendar-creation/oncall-calendar-creation.ipynb).

### CODE REVIEW

**Comment by anorth** [comment]
This makes it a lot more concrete for me, thanks. 

Not sure if this is the right repo for this to land in (it's public).

File: interviewing/take-home-assignments/oncall-calendar-creation/REPORT.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+I'm assuming you have an environment where can read/run .ipynb notebooks (e.g., JupyterLab, Visual Studo Code, PyCharm).
+
+## Download/Store Input Data
+``oncall-calendar-creation.ipynb`` expects a file in the same directory named ``Oncall Calendar Creation Input Data.xlsx``.  This file was created by exporting the Google Sheet input data as an .xlsx file.
+
+## GCP Setup
+Note: these steps are based on https://google-calendar-simple-api.readthedocs.io/en/latest/getting_started.html and https://developers.google.com/calendar/quickstart/python
+1. [Create a GCP project](https://developers.google.com/workspace/guides/create-project)
+2. [Enable the Google Calendar API](https://developers.google.com/workspace/guides/create-project#enable-api)
+3. [Generate Authroization credentials for a desktop application](https://developers.google.com/workspace/guides/create-credentials)
```
Comment (mxinden) [inline interviewing/take-home-assignments/oncall-calendar-creation/REPORT.md:21]:
```suggestion
3. [Generate authorization credentials for a desktop application](https://developers.google.com/workspace/guides/create-credentials)
```

File: interviewing/take-home-assignments/oncall-calendar-creation/REPORT.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+# Time Spent
+I recorded the steps I took and the time spent in ``worklog.md``.
+
+# Output
+The created Google Calendar id from this excercise is ``c_o8vr815a5uqg6kelc4oj1r7ek0@group.calendar.google.com``.
+It can also be accessed at https://calendar.google.com/calendar/u/0/embed?src=c_o8vr815a5uqg6kelc4oj1r7ek0@group.calendar.google.com&ctz=America/Los_Angeles
+
+# Areas To Improve
+If I had more time, I would invest in itmes like:
```
Comment (mxinden) [inline interviewing/take-home-assignments/oncall-calendar-creation/REPORT.md:39]:
```suggestion
If I had more time, I would invest in items like:
```

File: interviewing/take-home-assignments/oncall-calendar-creation/worklog.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+2021-06-10
+17:00 - created this file
+17:15 - got hello world working that can ready my calendar.  This involves GCP setup and getting related crentials.
```
Comment (mxinden) [inline interviewing/take-home-assignments/oncall-calendar-creation/worklog.md:3]:
```suggestion
17:15 - got hello world working that can read my calendar.  This involves GCP setup and getting related crentials.
```

**Review by mxinden — APPROVED** [review]
Looks good to me as a sample solution, though keep in mind that I am a Python novice.

> A rendered view of the notebook can be found here.

Rendered notebook is a great delivery format.

> Not sure if this is the right repo for this to land in (it's public).

Same.

**Comment by BigLep** [comment]
Thanks @mxinden and @anorth for the review.

Complete brain misfire to forget that this repo was public.  Doh!  I'll close this PR and put it against https://github.com/protocol/w3dt-stewards (which is private) and merge there.

Thanks for the suggested fixes Max.  I'll get those incorporated.

**Comment by BigLep** [comment]
Moved to https://github.com/protocol/w3dt-stewards/pull/2

**Pull request closed without merging**